quarkus/service/src/test/java/org/apache/polaris/service/quarkus/config/DefaultConfigurationStoreTest.java

```diff
@@ -111,4 +168,46 @@ public void testGetRealmConfiguration() {
     String keyTwoRealm3 = defaultConfigurationStore.getConfiguration(realm3Ctx, "key2");
     assertThat(keyTwoRealm3).isEqualTo(defaultKeyTwoValue);
   }
+
+  // TODO simplify once DefaultConfigrationStore doesn't rely on CallContext.getCurrentContext
+  private void setCurrentRealm(String realmName) {
+    RealmContext realmContext = () -> realmName;
+    QuarkusMock.installMockForType(realmContext, RealmContext.class);
+    CallContext.setCurrentContext(
+        new CallContext() {
+          @Override
+          public RealmContext getRealmContext() {
+            return realmContext;
+          }
+
+          @Override
+          public PolarisCallContext getPolarisCallContext() {
+            return CallContext.getCurrentContext().getPolarisCallContext();
+          }
+
+          @Override
+          public Map<String, Object> contextVariables() {
+            return CallContext.getCurrentContext().contextVariables();
+          }
+        });
+  }
+
+  @Test
+  public void testInjectedConfigurationStore() {
+    // Feature override makes this `false`
+    boolean featureOverrideValue =
```

Comment:
Nit: a possible enhancement would be to also inject `PolarisConfiguration` and check the contents of `defaults()` and `realmOverrides()` directly.